znnl/training_strategies/training_steps.py

```diff
@@ -0,0 +1,173 @@
+"""
+ZnNL: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+"""
+
+import logging
+from functools import partial
+from typing import Callable
+
+import jax.random
+from flax.training.train_state import TrainState
+from jax import jit, lax
+
+logger = logging.getLogger(__name__)
+
+
+class TrainStep:
```

Comment:
Can you add some doc strings here? I think we should also have a test for this class as far as it is possible. It is called in the `training_strategies` test but it's always better to unit test stuff like this and make sure nothing is broken.

Reply:
Done in [0fd55b7](https://github.com/zincware/ZnNL/pull/105/commits/0fd55b7fe1343d50d2ae0d4bfaedeeb98359096e) and [9b8021e](https://github.com/zincware/ZnNL/pull/105/commits/9b8021e72812c84f815f5723a84833320976c533)